src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IBinaryOperatorExpression.cs

```diff
@@ -898,5 +903,1194 @@ void M(bool a, bool b, bool c, bool d, bool e)
 
             VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedGraph, expectedDiagnostics);
         }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void LogicalFlow_06()
+        {
+            string source = @"
+using System;
+class C
+{
+    void M(int? a, int b)
+    /*<bind>*/{
+        GetArray()[0] = (a ?? b) + b;
+    }/*</bind>*/
+
+    static int[] GetArray() => null;
+}
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            string expectedOperationTree = @"
+IBlockOperation (1 statements) (OperationKind.Block, Type: null) (Syntax: '{ ... }')
+  IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'GetArray()[ ...  ?? b) + b;')
+    Expression: 
+      ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32) (Syntax: 'GetArray()[ ... a ?? b) + b')
+        Left: 
+          IArrayElementReferenceOperation (OperationKind.ArrayElementReference, Type: System.Int32) (Syntax: 'GetArray()[0]')
+            Array reference: 
+              IInvocationOperation (System.Int32[] C.GetArray()) (OperationKind.Invocation, Type: System.Int32[]) (Syntax: 'GetArray()')
+                Instance Receiver: 
+                  null
+                Arguments(0)
+            Indices(1):
+                ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 0) (Syntax: '0')
+        Right: 
+          IBinaryOperation (BinaryOperatorKind.Add) (OperationKind.BinaryOperator, Type: System.Int32) (Syntax: '(a ?? b) + b')
+            Left: 
+              ICoalesceOperation (OperationKind.Coalesce, Type: System.Int32) (Syntax: 'a ?? b')
+                Expression: 
+                  IParameterReferenceOperation: a (OperationKind.ParameterReference, Type: System.Int32?) (Syntax: 'a')
+                ValueConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+                  (Identity)
+                WhenNull: 
+                  IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.Int32) (Syntax: 'b')
+            Right: 
+              IParameterReferenceOperation: b (OperationKind.ParameterReference, Type: System.Int32) (Syntax: 'b')
+";
+            VerifyOperationTreeAndDiagnosticsForTest<BlockSyntax>(source, expectedOperationTree, expectedDiagnostics);
```

Comment:
>VerifyOperationTreeAndDiagnosticsForTest<BlockSyntax>(source, expectedOperationTree, expectedDiagnostics); [](start = 12, length = 106)

It is probably an overkill to verify original IOperation tree for the entire block. Is there a particular reason why you do this? I did some verification for Coalesce expression specifically because, frankly, we didn't have test coverage for them and I was making changes to the ICoalesceOperation node itself. If for any reason you would like to keep this verification, consider narrowing verification to a more limited set of nodes and please do this on the same compilation.

Reply:
Removed the extraneous tests.

---
In reply to: [164225986](https://github.com/dotnet/roslyn/pull/24465#discussion_r164225986) [](ancestors = 164225986)